tokio-executor/src/thread_pool/set.rs

```diff
@@ -0,0 +1,179 @@
+            shared.push(Shared::new(unpark));
+            owned.push(UnsafeCell::new(CachePadded::new(Owned::new(queue, rand))));
+        }
+
+        Set {
+            shared: shared.into_boxed_slice(),
+            owned: owned.into_boxed_slice(),
+            inject,
+            idle: Idle::new(num_workers),
+        }
+    }
+
+    fn inject_task(&self, task: Task<Shared<P>>) {
+        self.inject.push(task);
+        self.notify_work();
+    }
+
+    pub(super) fn notify_work(&self) {
+        if let Some(index) = dbg!(self.idle.worker_to_notify()) {
+            self.shared[index].unpark();
+        }
+    }
+
+    pub(super) fn notify_all(&self) {
+        for shared in &self.shared[..] {
+            shared.unpark();
+        }
+    }
+
+    pub(crate) fn spawn_background<F>(&self, future: F)
+    where
+        F: Future + Send + 'static,
+        F::Output: Send + 'static,
+    {
+        let task = task::background(future);
+        self.schedule(task);
+    }
+
+    pub(crate) fn schedule(&self, task: Task<Shared<P>>) {
+        current::get(|current_worker| match current_worker.as_member(self) {
+            Some(worker) => {
+                if dbg!(worker.submit_local(task)) {
+                    self.notify_work();
+                }
+            }
+            None => {
+                dbg!("Set::schedule - inject_task");
+                self.inject_task(task);
+            }
+        })
+    }
+
+    pub(crate) fn set_container_ptr(&mut self) {
+        dbg!(("set_container_ptr", self as *const _ as usize));
+
+        let ptr = self as *const _;
+        for shared in &mut self.shared[..] {
+            shared.set_container_ptr(ptr);
+        }
+    }
+
+    /// Signal the pool is closed
+    ///
+    /// Returns `true` if the transition to closed is successful. `false`
+    /// indicates the pool was already closed.
+    pub(crate) fn close(&self) -> bool {
+        if self.inject.close() {
+            self.notify_all();
+            true
+        } else {
+            false
+        }
+    }
+
+    pub(crate) fn len(&self) -> usize {
+        self.shared.len()
+    }
+
+    pub(super) fn index_of(&self, shared: &Shared<P>) -> usize {
+        use std::mem;
+
+        let size = mem::size_of::<Shared<P>>();
+
+        ((shared as *const _ as usize) - (&self.shared[0] as *const _ as usize)) / size
+    }
+
+    pub(super) fn shared(&self) -> &[Shared<P>] {
+        &self.shared
+    }
+
+    pub(super) fn owned(&self) -> &[UnsafeCell<CachePadded<Owned<P>>>] {
+        &self.owned
+    }
+
+    pub(super) fn idle(&self) -> &Idle {
+        &self.idle
+    }
+}
+
+impl Set<Box<dyn Unpark>> {
+    pub(crate) fn spawn_typed<F>(&self, future: F) -> JoinHandle<F::Output>
+    where
+        F: Future + Send + 'static,
+        F::Output: Send + 'static,
+    {
+        let (task, handle) = task::joinable(future);
+        self.schedule(task);
+        JoinHandle::new(handle)
+    }
+}
+
+impl<'a, P> Executor for &'a Set<P>
```

Comment:
```suggestion
impl<P> Executor for &Set<P>
```
nit: lifetime can be elided